src/go/rpk/pkg/cli/cmd/redpanda/admin/config/config.go

```diff
@@ -0,0 +1,237 @@
+to a logger persists for a limited amount of time, so as to ensure you do
+not accidentally enable debug logging permanently.
+
+It is optional to specify a logger; if you do not, this command will prompt
+from the set of available loggers.
+
+The special logger "all" enables all loggers. Alternatively, you can specify
+many loggers at once. To see all possible loggers, run the following command:
+
+  redpanda --help-loggers
+
+This command accepts loggers that it does not know of to ensure you can
+independently update your redpanda installations from rpk. The success or
+failure of enabling each logger is individually printed.
+`,
+
+		Run: func(_ *cobra.Command, loggers []string) {
+			cl := c.pickAdminAPI(host)
+
+			switch len(loggers) {
+			case 0:
+				choices := append([]string{"all"}, possibleLoggers...)
+				pick, err := out.Pick(choices, "Which logger would you like to set (all selects everything)?")
+				out.MaybeDie(err, "unable to pick logger: %v", err)
+				if pick == "all" {
+					loggers = possibleLoggers
+				} else {
+					loggers = []string{pick}
+				}
+
+			case 1:
+				if loggers[0] == "all" {
+					loggers = possibleLoggers
+				}
+			}
+
+			type failure struct {
+				logger string
+				err    error
+			}
+			var failures []failure
+			var successes []string
+
+			for _, logger := range loggers {
+				err := cl.SetLogLevel(logger, level, expirySeconds)
+				if err != nil {
+					failures = append(failures, failure{logger, err})
+				} else {
+					successes = append(successes, logger)
+				}
+			}
+
+			if len(successes) > 0 {
+				fmt.Println("SUCCESSES")
+				for _, success := range successes {
+					fmt.Println(success)
+				}
+				fmt.Println()
+			}
+			if len(failures) > 0 {
+				fmt.Println("FAILURES")
+				for _, failure := range failures {
+					fmt.Printf("%s: %v\n", failure.logger, failure.err)
+				}
+				fmt.Println()
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&level, "level", "l", "debug", "log level to set (error, warn, info, debug, trace)")
+	cmd.Flags().IntVarP(&expirySeconds, "expiry-seconds", "e", 300, "seconds to persist this log level override before redpanda reverts to its previous settings (if 0, persist until shutdown)")
+
+	cmd.Flags().StringVar(&host, "host", "", "either an index into admin_api hosts to issue the request to, or a hostname")
+	cobra.MarkFlagRequired(cmd.Flags(), "host")
+
+	return cmd
+}
+
+// List of possible loggers to set; more can be added in the future.
+// To generate this list, run redpanda --help-loggers
+var possibleLoggers = []string{
+	"admin_api_server",
+	"archival",
+	"assert",
+	"cloud_storage",
+	"cluster",
+	"compaction_ctrl",
+	"compression",
+	"coproc",
+	"dns_resolver",
+	"exception",
+	"fault_injector",
+	"http",
+	"httpd",
+	"io",
+	"kafka",
+	// "kafka/client", disabled until redpanda supports slashes in the handler
+	"kvstore",
+	"pandaproxy",
+	// "r/heartbeat", disabled until redpanda supports slashes in the handler
```

Comment:
@twmb - this patch could include doing a `r::heartbeat`  and `kafka::client` in C++ `::` the scope operator is used to denote namespaces, so is natural for a logger to have that too. 

Definitely not blocker of this patch, but *could*

Reply:
I just tried that locally—r::heartbeat does not trigger the r/heartbeat
logger.

fwiw this does support redpanda::main. My guess is that the http router is
interpreting the slash as a path separator and then cannot match the route.
I tried path escaping (as is in this code) but that also does not work.